SCons/Defaults.py

```diff
@@ -231,55 +242,84 @@ def chmod_func(dest, mode):
                     os.chmod(str(element), curr_perm & ~new_perm)
 
 
-def chmod_strfunc(dest, mode):
-    import SCons.Util
+def chmod_strfunc(dest, mode) -> str:
+    """strfunction for the Chmod action function."""
     if not SCons.Util.is_String(mode):
-        return 'Chmod(%s, 0%o)' % (get_paths_str(dest), mode)
+        return f'Chmod({get_paths_str(dest)}, {mode:#o})'
     else:
-        return 'Chmod(%s, "%s")' % (get_paths_str(dest), str(mode))
+        return f'Chmod({get_paths_str(dest)}, "{mode}")'
+
 
 
 Chmod = ActionFactory(chmod_func, chmod_strfunc)
 
 
-def copy_func(dest, src, symlinks=True):
-    """
-    If symlinks (is true), then a symbolic link will be
+
+def copy_func(dest, src, symlinks=True) -> int:
+    """Implementation of the Copy action function.
+
+    Copies *src* to *dest*.  If *src* is a list, *dest* must be
+    a directory, or not exist (will be created).
+
+    Since Python :mod:`shutil` methods, which know nothing about
+    SCons Nodes, will be called to perform the actual copying,
+    args are converted to strings first.
+
+    If *symlinks* evaluates true, then a symbolic link will be
     shallow copied and recreated as a symbolic link; otherwise, copying
     a symbolic link will be equivalent to copying the symbolic link's
     final target regardless of symbolic link depth.
     """
 
     dest = str(dest)
-    src = str(src)
+    src = [str(n) for n in src] if SCons.Util.is_List(src) else str(src)
 
     SCons.Node.FS.invalidate_node_memos(dest)
-    if SCons.Util.is_List(src) and os.path.isdir(dest):
+    if SCons.Util.is_List(src):
+        if not os.path.exists(dest):
+            os.makedirs(dest, exist_ok=True)
+        elif not os.path.isdir(dest):
+            # is Python's NotADirectoryError more appropriate?
+            raise SCons.Errors.UserError(
+                f'Copy() called with list src but dest "{dest}" is not a directory'
+            )
+
         for file in src:
             shutil.copy2(file, dest)
         return 0
+
     elif os.path.islink(src):
         if symlinks:
-            return os.symlink(os.readlink(src), dest)
-        else:
-            return copy_func(dest, os.path.realpath(src))
+            os.symlink(os.readlink(src), dest)
```

Comment:
why not return os.symlinks()'s return value?

Reply:
was trying to be a bit more consistent -` os.symlink` doesn't actually return (aka None) and this *seems* to want to return 0 on success.  except I'm far from convinced anything looks at the return from one of these functions the way they're embedded in whatever an ActionFactor sets up.

Reply:
Gotcha.